main.cpp

```diff
@@ -92,7 +92,81 @@ good to go!
  */
 
 #include <iostream>
+
+struct FloatType
+{
+    float add( float lhs, float rhs);
+    float subtract( float lhs, float rhs);
+    float multiply( float lhs, float rhs);
+    float divide( float lhs, float rhs);
+};
+
+float FloatType::add ( float lhs, float rhs) { return lhs + rhs; }
+float FloatType::subtract ( float lhs, float rhs) { return lhs - rhs; }
+float FloatType::multiply ( float lhs, float rhs) { return lhs * rhs; }
+float FloatType::divide ( float lhs, float rhs)
+{
+    if( rhs == 0.0f ) 
+    {
+        std::cout << std::endl;
+        std::cout << "warning, floating point division by zero returns 'inf' !" << std::endl;
+    }
+    return lhs / rhs;
+}
+
+struct DoubleType
+{
+    double add( double lhs, double rhs);
+    double subtract( double lhs, double rhs);
+    double multiply( double lhs, double rhs);
+    double divide( double lhs, double rhs);
+};
+
+double DoubleType::add ( double lhs, double rhs) { return lhs + rhs; }
+double DoubleType::subtract ( double lhs, double rhs) { return lhs - rhs; }
+double DoubleType::multiply ( double lhs, double rhs) { return lhs * rhs; }
+double DoubleType::divide ( double lhs, double rhs) 
+{
+    if( rhs > 0 ) 
+    {
+        return lhs / rhs;
+    }
+    else
+    {    
+        std::cout << std::endl;
+        std::cout << "warning, floating point division by zero returns 'inf' !" << std::endl;
+    }
+    return lhs / rhs;
```

Comment:
you wrote `return lhs/rhs;` twice.  Also, You're only checking if rhs is GREATER than 0.  What is RHS is negative?
According to the logic written, a warning about a divide-by-zero will be emitted.   is that the correct logic?
- [ ] revise